mailbox/api/src/main/java/org/apache/james/mailbox/events/EventDeadLetters.java

```diff
@@ -0,0 +1,33 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.events;
+
+import org.apache.james.mailbox.Event;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+interface EventDeadLetters {
+    Mono<Void> store(Group registeredGroup, Event failDeliveredEvent);
+
+    Flux<Event> take(Group registeredGroup);
```

Comment:
`take` seems like a bad name IMO

Why not `listFailedEvents`?

Reply:
Moreover, we should be able to : 
 - List events ids failed for a group without removing them
 - Getting an event with group + EventId
 - Deleting a single event (group + eventId)